cmd/overlay.go

```diff
@@ -93,14 +93,18 @@ func runCompare(ctx context.Context, flags overlayCompareFlags) error {
 
 func runApply(ctx context.Context, flags overlayApplyFlags) error {
 	out := os.Stdout
+	yamlOut := true
+
 	if flags.Out != "" {
 		file, err := os.Create(flags.Out)
 		if err != nil {
 			return err
 		}
 		defer file.Close()
 		out = file
+
+		yamlOut = utils.HasYAMLExt(flags.Out)
```

Comment:
Wondering if we should flip this and look for json extensions. i.e. if I just put a file name like output or output.txt we'd write yaml to it by default rather than json